acados/ocp_nlp/ocp_nlp_common.h

```diff
@@ -89,13 +89,16 @@ typedef struct
     void (*opts_set_at_stage)(void *config_, void *opts_, int stage, const char *field, void* value);
     // evaluate solver // TODO rename into solve
     int (*evaluate)(void *config, void *dims, void *nlp_in, void *nlp_out, void *opts_, void *mem, void *work);
-    void (*eval_param_sens)(void *config, void *dims, void *opts_, void *mem, void *work, char *field, int stage, int index, void *sens_nlp_out);
+    void (*eval_param_sens)(void *config, void *dims, void *opts_, void *mem, void *work,
+                            char *field, int stage, int index, void *sens_nlp_out);
     // prepare memory
     int (*precompute)(void *config, void *dims, void *nlp_in, void *nlp_out, void *opts_, void *mem, void *work);
     // initalize this struct with default values
     void (*config_initialize_default)(void *config);
     // general getter
     void (*get)(void *config_, void *dims, void *mem_, const char *field, void *return_value_);
+    void (*opts_get)(void *config_, void *dims, void *opts_, const char *field, void *return_value_);
+    void (*work_get)(void *config_, void *dims, void *work_, const char *field, void *return_value_);
```

Comment:
why do you need to get opts and work of the nlp?

Reply:
I needed them here:
https://github.com/acados/acados/blob/3ac5ab40a79ada831e0c6f8d9e2c2617f3a04bab/interfaces/acados_c/ocp_nlp_interface.c#L913-L926

Reply:
I was just wondering why get opts and work has never been necessary, probably there is another way of doing it without using them.

Reply:
I guess the other way of doing it would be to do it inside the sqp and sqp_rti solvers, instead of exposing the nlp_opts, nlp_mem, nlp_work as stand-alone.

Reply:
exactly, the alternative was to duplicate the code for the function I mentioned for SQP and SQP_RTI.

Reply:
Not necessarily. The function could be implemented once in nlp_common, and there would just be two wrappers in SQP and SQP_RTI.